widgetssdk/src/main/java/com/glia/widgets/helper/ActivityWatcher.kt

```diff
@@ -0,0 +1,49 @@
+package com.glia.widgets.helper
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import androidx.annotation.VisibleForTesting
+import com.glia.widgets.callvisualizer.controller.CallVisualizerController
+import java.lang.ref.WeakReference
+
+class ActivityWatcher(
+    private val app: Application,
+    private val callVisualizerController: CallVisualizerController
```

Comment:
I'm not sure, but maybe ActivityWatcher shouldn't work directly with the controller. I also don't like that CallVisualizerController works with Activity references.

Reply:
Usually, `Controller` is included to an `Activity` or `View` class as a variable. Please let me know if I'm wrong though. Since Call Visualizer should display its dialogs on top of any `Activity`, I am looking at `ActivityWatcher`as a replacement for `Activity` or `View` here.

`CallVisualizerController` doesn't hold an `Activity` reference, it's only accepting `Activity` as a parameter to its functions. So, it shouldn't be the case for any kind of memory leak, should it?

If you still have any concerns, could you please suggest alternatives? How do you see it in a different way?

Reply:
So, this controller will use only with activity watcher? Maybe it's OK.

Reply:
In this case, it would be better to show the dialog from the `ActivityWatcher` and name it another way, e.g. `DialogHolder`, `DialogWrapper`, and it can use context directly from `WeakReference`

Reply:
@DavDo @AndriiHorishniiMOC I've made changes, please have a look now.